hummingbot/strategy/avellaneda_market_making/avellaneda_market_making.pyx

```diff
@@ -631,7 +634,11 @@ cdef class AvellanedaMarketMakingStrategy(StrategyBase):
                 self.c_cancel_active_orders_on_max_age_limit()
                 self.c_cancel_active_orders(proposal)
 
-                self._execution_state.process_tick(timestamp, self)
+                is_processed = self._execution_state.process_tick(timestamp, self)
+
+                # Finalize time left if no trading is allowed anymore
+                if not is_processed:
+                    self._time_left = 0
```

Comment:
It seems like `_time_left` is something that should be handle by `_execution_state`.